packages/tools/client-logger/app-insights-logger/src/appInsightsLogger.ts

```diff
@@ -0,0 +1,42 @@
+import { ApplicationInsights, Snippet } from "@microsoft/applicationinsights-web";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+export interface AppInsightsLoggerConfig {
+	appInsightsClient?: ApplicationInsights;
+	appInsightsConfig?: Snippet;
+}
+
+export class AppInsightsLogger implements ITelemetryBufferedLogger {
+	protected readonly baseLoggingClient: ApplicationInsights;
+
+	public constructor(loggerConfig: AppInsightsLoggerConfig) {
+		if (loggerConfig.appInsightsClient) {
+			this.baseLoggingClient = loggerConfig.appInsightsClient;
+		} else {
+			if (loggerConfig.appInsightsConfig === undefined) {
+				throw Error(
+					"Cannot initialize default AppInsights Client without a configuration object.",
+				);
+			}
+			this.baseLoggingClient = new ApplicationInsights(loggerConfig.appInsightsConfig);
+			this.baseLoggingClient.loadAppInsights();
+		}
+	}
+
+	getBaseLoggingClient() {
```

Comment:
Do we need this exposed?

Reply:
I think we should lean the other way, not expose it until we have a specific need for it. Removing something we exposed before (intentionally or not) is always harder than exposing new things as the need arises. In the case where the consumer provided the App Insights client, they already have it anyway. And if they provided just the config Snippet instead, my guess is that they probably don't need it (or they would have instantiated it themselves).